sky/data/storage.py

```diff
@@ -1150,6 +1202,48 @@ def _validate(self):
                     assert data_utils.verify_r2_bucket(self.name), (
                         f'Source specified as {self.source}, a R2 bucket. ',
                         'R2 Bucket should exist.')
+        # Validate name
+        self._validate_name(self.name)
+
+    @classmethod
+    def _validate_name(cls, name) -> None:
```

Comment:
We need to check for the following cases as well for GCS:
<img width="538" alt="Screenshot 2023-04-07 at 8 33 24 PM" src="https://user-images.githubusercontent.com/34902420/230701396-8087a201-9213-437d-b7bb-0b9b3ea3bbfd.png">

Reply:
Thanks! Fixed now.